packages/flutter/lib/src/material/menu_style.dart

```diff
@@ -0,0 +1,447 @@
+  ///    within a [Theme].
+  final VisualDensity? visualDensity;
+
+  /// Determines the desired alignment of the submenu when opened relative to
+  /// the button that opens it.
+  ///
+  /// If there isn't sufficient space to open the menu with the given alignment,
+  /// and there's space on the other side of the button, then the alignment is
+  /// swapped to it's opposite (1 becomes -1, etc.), and the menu will try to
+  /// appear on the other side of the menu. If there isn't enough space there
+  /// either, then the menu will be pushed as far over as necessary to display
+  /// as much of itself as possible, possibly overlapping the parent button.
+  ///
+  /// Defaults to [AlignmentDirectional.topEnd], which means that in a
+  /// left-to-right context, the menu will appear with its origin at the upper
+  /// right corner of the [MenuButton] that owns it. This applies for vertical
+  /// menus (e.g. created by a [MenuButton] or [createMaterialMenu]).
+  ///
+  /// For a [MenuBar], the default is [AlignmentDirectional.bottomStart].
+  final AlignmentGeometry? alignment;
+
+  /// Returns a copy of this MenuStyle with the given fields replaced with
+  /// the new values.
+  MenuStyle copyWith({
+    MaterialStateProperty<Color?>? backgroundColor,
+    MaterialStateProperty<Color?>? shadowColor,
+    MaterialStateProperty<Color?>? surfaceTintColor,
+    MaterialStateProperty<double?>? elevation,
+    MaterialStateProperty<EdgeInsetsGeometry?>? padding,
+    MaterialStateProperty<Size?>? minimumSize,
+    MaterialStateProperty<Size?>? fixedSize,
+    MaterialStateProperty<Size?>? maximumSize,
+    MaterialStateProperty<BorderSide?>? side,
+    MaterialStateProperty<OutlinedBorder?>? shape,
+    MaterialStateProperty<MouseCursor?>? mouseCursor,
+    VisualDensity? visualDensity,
+    AlignmentGeometry? alignment,
+  }) {
+    return MenuStyle(
+      backgroundColor: backgroundColor ?? this.backgroundColor,
+      shadowColor: shadowColor ?? this.shadowColor,
+      surfaceTintColor: surfaceTintColor ?? this.surfaceTintColor,
+      elevation: elevation ?? this.elevation,
+      padding: padding ?? this.padding,
+      minimumSize: minimumSize ?? this.minimumSize,
+      fixedSize: fixedSize ?? this.fixedSize,
+      maximumSize: maximumSize ?? this.maximumSize,
+      side: side ?? this.side,
+      shape: shape ?? this.shape,
+      mouseCursor: mouseCursor ?? this.mouseCursor,
+      visualDensity: visualDensity ?? this.visualDensity,
+      alignment: alignment ?? this.alignment,
+    );
+  }
+
+  /// Returns a copy of this object, but resets to null any properties passed as
+  /// true.
+  ///
+  /// For instance, if you want to have a [MenuStyle] that is the same as
+  /// another style, but reset the `backgroundColor` to null so that it uses the
+  /// default color, you can call:
+  ///
+  /// ```dart
+  /// MenuStyle newStyle = style.copyWithout(backgroundColor: true);
+  /// ```
+  MenuStyle copyWithout({
```

Comment:
For the sake of consistency we should a copyWithout method to ButtonStyle (in a separate PR).